### PR TITLE
Refactor auth flow

### PR DESCRIPTION
**Pull request by gkang2018**
Added necessary changes to organizational signup flow per the following issues, #247 and #189 

1. Eliminate redundant looks great validations on all the signup fields, now it's just the check mark 
2. Got rid of the "one or more invalid" error when you go back on the personal information page
3. Must fill in organizational website field
4. Persist signature when navigating to other parts of the form

### CODE REVIEW

**Review by crchong1 — APPROVED** [review]
looks great dude- i like it

**Comment by crchong1** [comment]
wait actually can you add 3-4 tests

**Review by crchong1 — CHANGES_REQUESTED** [review]
see above

**Review by crchong1 — APPROVED** [review]
looks fantastic- i think all of these tests are great!

File: src/tests/components/SignUp/SignUserAgreement.test.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import React from 'react';
+import { Provider, transitions } from 'react-alert';
+import AlertTemplate from 'react-alert-template-basic';
+import { MemoryRouter } from 'react-router-dom';
+
+import { SignUserAgreement } from '../../../components/SignUp/pages/SignUserAgreement';
+
+const options = {
+  position: 'bottom left',
+  timeout: 5000,
```
Comment (crchong1) [inline src/tests/components/SignUp/SignUserAgreement.test.tsx:13]: do we need this?

Comment (gkang2018) [reply src/tests/components/SignUp/SignUserAgreement.test.tsx:13]: i think so, i talked w nick about this. Essentially, a lot of these tests fail when we try to render the component itself without the provider tag, so our work around was that we render the provider tag as well and these are the props you need for it.

Comment (gkang2018) [reply src/tests/components/SignUp/SignUserAgreement.test.tsx:13]: yeah i just took a look at this and i guess we only need the provider for the CompleteSignupflow test and not the other components, good catch

**Pull request merged**